src/compiler/checker.ts

```diff
@@ -7433,7 +7433,7 @@ namespace ts {
                     skippedPrivateMembers.set(rightProp.name, true);
                 }
                 else if (!isClassMethod(rightProp) && !isSetterWithoutGetter) {
-                    members.set(rightProp.name, rightProp);
+                    members.set(rightProp.name, getNonReadonlySymbol(rightProp));

```

Comment:
this is going to make unique symbols for each spread that uses the original property. Probably there should be some caching to prevent that. Maybe it would be a good idea to start caching the spread object types at this point; it would accomplish the same thing.